pkg/cmd/server/startconfig/kube_connection_args.go

```diff
@@ -0,0 +1,117 @@
+package startconfig
+
+import (
+	"errors"
+	"net/url"
+
+	"github.com/spf13/pflag"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+
+	"github.com/openshift/origin/pkg/cmd/flagtypes"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+)
+
+type KubeConnectionArgs struct {
+	KubernetesAddr flagtypes.Addr
+
+	// ClientConfig is used when connecting to Kubernetes from the master, or
+	// when connecting to the master from a detached node. If StartKube is true,
+	// this value is not used.
+	ClientConfig clientcmd.ClientConfig
+	// ClientConfigLoadingRules is the ruleset used to load the client config.
+	// Only the CommandLinePath is expected to be used.
+	ClientConfigLoadingRules clientcmd.ClientConfigLoadingRules
+
+	CertArgs *CertArgs
```

Comment:
Is this used when we are starting kube in-process and populating with our kube-client certs?